### ISSUE TITLE
Building application / library should verify proper project target version

### ISSUE THREAD

**Issue by per-samuelsson** [Starcounter/Starcounter#4143]
The other side of https://github.com/Starcounter/Starcounter.VisualStudio/issues/1. Will verify templates / consuming products are in sync with the installed runtime.

**Comment by per-samuelsson** [comment]
If things work out OK, I'll push this today, and tweak it and merge it into develop after that.

Here's how it works:

In all 2.3-based versions, when a project targeting Starcounter is *built*, we verify that the project either is explicitly tagged to target that platform version. That's done in the project file, using:

```xml
<StarcounterVersionCompatibility>2.3</StarcounterVersionCompatibility>
```

The be backwards compatible, versions in the 2.3 range treat any project that does *not* have this property set at all as compatible with 2.3 and hence grant the build.

If the compatibility level is something else, you'll get an error such as this:

![image](https://cloud.githubusercontent.com/assets/2902045/23947342/a9b8c10c-097e-11e7-940e-720cae2e687e.png)

In 2.4 (and above), we'll *force* the compatibility to be set and to be 2.4 exclusively. Meaning that if you accidentally bring a project from 2.3 to a 2.4 installation, you'll get an error similar to that above.

Reminder why we need this: we've changed how projects must reference Starcounter from 2.4 and onward, with the introduction of build-time weaving. That's also the reason why we need two versions of the extension, one pre-2.4 and one starting from that.

**Comment by per-samuelsson** [comment]
Done. Merged and pushed to 2.3 in https://github.com/Starcounter/Starcounter/commit/1120af8965b1a2c0d7cd9b61007e37f75aa90766, and thereafter merged into `develop` and adapted to that context.
